File: go/etl/etl.go
// Package etl implements the exercism.io Transform solution
package etl

import (
	"strings"
)

// Transform reformats the input map into the result, mapping the input key as the value of each element of the result
func Transform(input map[int][]string) map[string]int {
	var result = make(map[string]int, 26)
	for points, letters := range input {
		for _, letter := range letters {
			result[strings.ToLower(letter)] = points
		}
	}

	return result
}

File: elixir/etl/lib/etl.ex
defmodule ETL do
  @doc """
  Transforms an old Scrabble score system to a new one.

  ## Examples

    iex> ETL.transform(%{1 => ["A", "E"], 2 => ["D", "G"]})
    %{"a" => 1, "d" => 2, "e" => 1, "g" => 2}
  """
  @spec transform(map) :: map
  def transform(input) do
    Enum.reduce(input, %{}, fn {score, letters}, result -> transform(score, letters, result) end)
  end

  defp transform(score, letters, result) do
    Enum.reduce(letters, result, fn letter, result -> Map.put(result, String.downcase(letter), score) end)
  end
end
